### PR TITLE
mat: relax list test type restrictions for Vector parameters

### PR DESCRIPTION
**Pull request by kortschak**
Please take a look.

### CODE REVIEW

File: mat/list_test.go [hunk 1]
```diff
@@ -169,14 +169,20 @@ func legalTypesSym(a, b Matrix) bool {
 	return true
 }
 
+// legalTypeVec returns whether v is a Vector.
```
Comment (vladimir-ch) [inline mat/list_test.go]: legalTypeVector

File: mat/list_test.go [hunk 2]
```diff
@@ -197,9 +203,16 @@ func legalTypesVecDenseVecDense(a, b Matrix) bool {
 	return true
 }
 
-// legalTypesNotVecVec returns whether the first input is an arbitrary Matrix
+// legalTypesNotVectorVector returns whether the first input is an arbitrary Matrix
+// and the second input is a Vector.
+func legalTypesNotVectorVector(a, b Matrix) bool {
```
Comment (vladimir-ch) [inline mat/list_test.go]: How about renaming to legalTypesMatrixVector?

File: mat/list_test.go [hunk 3]
```diff
@@ -197,9 +203,16 @@ func legalTypesVecDenseVecDense(a, b Matrix) bool {
...
+	_, ok := b.(Vector)
+	return ok
+}
+
+// legalTypesNotVectorVecDense returns whether the first input is an arbitrary Matrix
 // and the second input is a *VecDense.
-func legalTypesNotVecVec(a, b Matrix) bool {
+func legalTypesNotVectorVecDense(a, b Matrix) bool {
```
Comment (vladimir-ch) [inline mat/list_test.go]: legalTypesMatrixVecDense?

File: mat/list_test.go (hunk 1 above)
Comment (kortschak) [reply mat/list_test.go]: Done

File: mat/list_test.go (hunk 2 above)
Comment (kortschak) [reply mat/list_test.go]: Thanks, I was looking for something like that.

File: mat/list_test.go (hunk 3 above)
Comment (kortschak) [reply mat/list_test.go]: Done

**Comment by kortschak** [comment]
PTAL

**Pull request merged**